cmd/kubeadm/app/constants/constants.go

```diff
@@ -208,8 +209,33 @@ var (
 
 	// MinimumKubeletVersion specifies the minimum version of kubelet which kubeadm supports
 	MinimumKubeletVersion = version.MustParseSemantic("v1.8.0")
+
+	// SupportedEtcdVersion lists officially supported etcd versions with corresponding kubernetes releases
+	SupportedEtcdVersion = map[string]string{
+		"v1.7": "3.0.17",
+		"v1.8": "3.0.17",
+		"v1.9": "3.1.10",
+	}
 )
 
+// GetEtcdSupportedVersion returns officially supported version of etcd for a specific kubernetes release
+// if passed version is not listed, the function returns empty string and an error
+func GetEtcdSupportedVersion(k8sVersion string) (string, error) {
```

Comment:
Pass in a `*version.Version` here instead and just return things from `SupportedEtcdVersion` based on `k8sVersion.Minor()`

Reply:
done a bit differently ;)